### ISSUE TITLE
TextField with inputFormatters and keyboardType

### ISSUE THREAD

**Issue by Lomski** [flutter/flutter#60568]
## Steps to Reproduce

TextField with inputFormatters (for example, WhitelistingTextInputFormatter.digitsOnly) and some of keyboardTypes (phone, datetime, visiblePassword and number) work not as it expected. If i type not a digit (remember, we are using WhitelistingTextInputFormatter.digitsOnly inputFormatters), onChange fires twice and the last one will be with forbidden symbol in onChange text. With another keyboardType onChange fires only once and without "forbidden" symbol in the text (not sure if it even should fire).

**Expected results:** 
Forbidden symbols shouldn't be in the onChange text.

**Actual results:** 
onChange fires twice with forbidden symbol in the last change.

**Comment by darshankawar** [comment]
Hi @Lomski,
Can you please provide your `flutter doctor -v`, your `flutter run --verbose` and a minimal complete reproducible code sample.
Thank you

**Comment by VladyslavBondarenko** [comment]
Duplicate of https://github.com/flutter/flutter/issues/61522
@Lomski 
it should be already fixed, please provide a minimal complete reproducible code sample if your case is different and I'll reopen

**Comment by Jinxishihenian** [comment]
> ## 重现步骤
> 带有inputFormatters（例如WhitelistingTextInputFormatter.digitsOnly）和某些keyboardTypes（电话，日期时间，visiblePassword和数字）的TextField不能正常工作。如果我键入的不是数字（请记住，我们使用的是WhitelistingTextInputFormatter.digitsOnly inputFormatters），则onChange将触发两次，最后一个将在onChange文本中带有禁止符号。使用另一个keyboardType时，onChange仅触发一次，并且在文本中没有“禁止”符号（不确定是否应该触发）。
> 
> **预期的结果：**
> 禁止的符号不应出现在onChange文本中。
> 
> **实际结果：**
> onChange在最后一次更改中使用禁止符号触发两次。

I have the same problem

**Comment by Lomski** [comment]
@VladyslavBondarenko 
```
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String defaultKeyboardText = '';
  String numberKeyboardText = '';

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: <Widget>[
          Padding(
            padding: const EdgeInsets.only(top: 40, left: 20),
            child: Text(
              'Default keyboard',
            ),
          ),
          TextField(
            inputFormatters: [WhitelistingTextInputFormatter.digitsOnly],
            onChanged: (text) {
              setState(() {
                defaultKeyboardText = text;
              });
            },
          ),
          Text.rich(TextSpan(
            children: [
              TextSpan(text: 'Default keyboard type text is: '),
              TextSpan(
                text: defaultKeyboardText,
                style: TextStyle(color: Colors.red, fontWeight: FontWeight.w600),
              )
            ],
          )),
          Padding(
            padding: const EdgeInsets.only(top: 40, left: 20),
            child: Text(
              'Number keyboard',
            ),
          ),
          TextField(
            inputFormatters: [WhitelistingTextInputFormatter.digitsOnly],
            keyboardType: TextInputType.number,
            onChanged: (text) {
              setState(() {
                numberKeyboardText = text;
              });
            },
          ),
          Text.rich(TextSpan(
            children: [
              TextSpan(text: 'Number keyboard type text is: '),
              TextSpan(
                text: numberKeyboardText,
                style: TextStyle(color: Colors.red, fontWeight: FontWeight.w600),
              )
            ],
          )),
        ],
      ),
    );
  }
}
```
flutter-doctor:
```
[✓] Flutter (Channel stable, v1.17.5, on Linux, locale ru_RU.UTF-8)
    • Flutter version 1.17.5 at /home/user/Programs/flutter
    • Framework revision 8af6b2f038 (3 weeks ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/user/Android/Sdk
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /home/user/Android/Sdk
    • Java binary at: /home/user/.local/share/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[!] Android Studio (version 4.0)
    • Android Studio at /home/user/.local/share/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

```

**Comment by VladyslavBondarenko** [comment]
@Lomski 
consider to switch to beta channel (current beta version is 1.20.0-7.2.pre), it's fixed there and later flutter versions with [FilteringTextInputFormatter](https://master-api.flutter.dev/flutter/services/FilteringTextInputFormatter-class.html), which you can use the same way `FilteringTextInputFormatter.digitsOnly`
